kvmagent/kvmagent/kvmagent.py

```diff
@@ -122,9 +120,42 @@ def wrap(*args, **kwargs):
             rsp.error = str(e)
             logger.warn(err)
             return jsonobject.dumps(rsp)
-        
     return wrap
 
+
+def deleteImage(path,type):
+    if (type == 'localstorage' or  type == 'smp'):
+        shell.call('rm -f %s' % path)
+        pdir = os.path.dirname(path)
+        linux.rmdir_if_empty(pdir)
+        logger.debug('successfully delete %s' % path)
+        if (path.endswith('.qcow2')):
+            imf2File = path.replace('.qcow2','.imf2')
+            shell.call('rm -f %s' % imf2File)
+            pdir = os.path.dirname(imf2File)
+            linux.rmdir_if_empty(pdir)
+            logger.debug('successfully delete %s' % imf2File)
+            imfFile = path.replace('.qcow2','.imf')
+            shell.call('rm -f %s' % imfFile)
+            pdir = os.path.dirname(imfFile)
+            linux.rmdir_if_empty(pdir)
+            logger.debug('successfully delete %s' % imfFile)
+    if (type == 'nfs' ):
```

Comment:
The 'type' check is unnecessary.  And the deletion logic for imf2/imf should be combined, like:

if (path.endswith('.qcow2')):                                                   
    imfFiles = [ path.replace(".qcow2", x) for x in [ ".imf", ".imf2"] ]        
    shell.call('rm -f %s' % string.join(imfFiles, ' ')

you need to import "string" to use "join".